<a href="https://colab.research.google.com/github/juansouz789/Projetos-de-DS/blob/main/classifica%C3%A7%C3%A3o_de_filmes.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## **Importando bibliotecas**

In [ ]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
import plotly.express as px
plt.style.use('ggplot')

## **Visualizando o Dataset**

Primeiro iremos carregar só a base de dados dos filmes que contém gênero, id e título.

In [ ]:
filmes = pd.read_csv('/content/movies.dat', delimiter = '::')
filmes

/usr/local/lib/python3.8/dist-packages/pandas/util/_decorators.py:311: ParserWarning: Falling back to the 'python' engine because the 'c' engine does not support regex separators (separators > 1 char and different from '\s+' are interpreted as regex); you can avoid this warning by specifying engine='python'.
  return func(*args, **kwargs)


0000008      Edison Kinetoscopic Record of a Sneeze (1894)  \
0            10                La sortie des usines Lumière (1895)   
1            12                      The Arrival of a Train (1896)   
2            25  The Oxford and Cambridge University Boat Race ...   
3            91                         Le manoir du diable (1896)   
4           131                           Une nuit terrible (1896)   
...         ...                                                ...   
37336  14499632                                22 vs. Earth (2021)   
37337  14527836                                    Recalled (2021)   
37338  14544192                          Bo Burnham: Inside (2021)   
37339  14735160                             Mum is Pregnant (2021)   
37340  14740904                              Juanes: Origen (2021)   

               Documentary|Short  
0              Documentary|Short  
1              Documentary|Short  
2                            NaN  
3                   Short|Horror  
4            Short|Comedy|Horror  
...                          ...  
37336  Animation|Short|Adventure  
37337     Drama|Mystery|Thriller  
37338         Comedy|Drama|Music  
37339                        NaN  
37340                Documentary  

[37341 rows x 3 columns]

Como essa base não possuí nome das colunas iremos definir abaixo.

In [ ]:
filmes.columns = ['ID', 'Título', 'Gênero']
filmes

ID                                             Título  \
0            10                La sortie des usines Lumière (1895)   
1            12                      The Arrival of a Train (1896)   
2            25  The Oxford and Cambridge University Boat Race ...   
3            91                         Le manoir du diable (1896)   
4           131                           Une nuit terrible (1896)   
...         ...                                                ...   
37336  14499632                                22 vs. Earth (2021)   
37337  14527836                                    Recalled (2021)   
37338  14544192                          Bo Burnham: Inside (2021)   
37339  14735160                             Mum is Pregnant (2021)   
37340  14740904                              Juanes: Origen (2021)   

                          Gênero  
0              Documentary|Short  
1              Documentary|Short  
2                            NaN  
3                   Short|Horror  
4            Short|Comedy|Horror  
...                          ...  
37336  Animation|Short|Adventure  
37337     Drama|Mystery|Thriller  
37338         Comedy|Drama|Music  
37339                        NaN  
37340                Documentary  

[37341 rows x 3 columns]

Agora iremos carregar a base de dados de avaliações.

In [ ]:
avaliações = pd.read_csv('/content/ratings.dat', delimiter = '::')
avaliações

/usr/local/lib/python3.8/dist-packages/pandas/util/_decorators.py:311: ParserWarning: Falling back to the 'python' engine because the 'c' engine does not support regex separators (separators > 1 char and different from '\s+' are interpreted as regex); you can avoid this warning by specifying engine='python'.
  return func(*args, **kwargs)


1  0114508   8  1381006850
0           2   499549   9  1376753198
1           2  1305591   8  1376742507
2           2  1428538   1  1371307089
3           3    75314   1  1595468524
4           3   102926   9  1590148016
...       ...      ...  ..         ...
906825  70781  9893250  10  1613857551
906826  70781  9898858   3  1585958452
906827  70782   172495  10  1587107015
906828  70782   414387  10  1587107852
906829  70783  1623205   6  1362832655

[906830 rows x 4 columns]

Esta base também não veio com o nome das colunas, iremos definir abaixo.

In [ ]:
avaliações.columns = ['Usuário', 'ID', 'Notas', 'Timestamp']
avaliações


Usuário       ID  Notas   Timestamp
0             2   499549      9  1376753198
1             2  1305591      8  1376742507
2             2  1428538      1  1371307089
3             3    75314      1  1595468524
4             3   102926      9  1590148016
...         ...      ...    ...         ...
906825    70781  9893250     10  1613857551
906826    70781  9898858      3  1585958452
906827    70782   172495     10  1587107015
906828    70782   414387     10  1587107852
906829    70783  1623205      6  1362832655

[906830 rows x 4 columns]

Agora vamos juntar essas duas bases de dados. Ambas tem o ID do filme como uma coluna em comum, então podemos utiliza-la para mesclar os dados.

In [ ]:
nova = pd.merge(filmes, avaliações, on = ['ID', 'ID'])
nova

ID                                             Título  \
0             10                La sortie des usines Lumière (1895)   
1             12                      The Arrival of a Train (1896)   
2             25  The Oxford and Cambridge University Boat Race ...   
3             91                         Le manoir du diable (1896)   
4             91                         Le manoir du diable (1896)   
...          ...                                                ...   
908627  14544192                          Bo Burnham: Inside (2021)   
908628  14544192                          Bo Burnham: Inside (2021)   
908629  14544192                          Bo Burnham: Inside (2021)   
908630  14735160                             Mum is Pregnant (2021)   
908631  14740904                              Juanes: Origen (2021)   

                    Gênero  Usuário  Notas   Timestamp  
0        Documentary|Short    70577     10  1412878553  
1        Documentary|Short    69535     10  1439248579  
2                      NaN    37628      8  1488189899  
3             Short|Horror     5814      6  1385233195  
4             Short|Horror    37239      5  1532347349  
...                    ...      ...    ...         ...  
908627  Comedy|Drama|Music    55241     10  1622416491  
908628  Comedy|Drama|Music    57060     10  1623092790  
908629  Comedy|Drama|Music    68109      3  1622844954  
908630                 NaN    19702      9  1622798273  
908631         Documentary    51324      6  1622959573  

[908632 rows x 6 columns]

## **Analise**

In [ ]:
nova.Notas.value_counts()

8     219311
7     203476
9     128749
6     118323
10    107284
5      68458
4      27779
3      15258
1      10663
2       9053
0        278
Name: Notas, dtype: int64

In [ ]:
av = nova['Notas'].value_counts()
numeros = av.index
quantidade = av.values
fig = px.pie(nova, values=quantidade, names=numeros)
fig.show()

De acordo com o gráfico a maioria dos filmes são classificados positivamente com nota 8.

Agora daremos uma olhada nos filmes que obtiveram a nota máxima(10).

In [ ]:
nova2 = nova.query('Notas==10')
print(nova2['Título'].value_counts().head(10))

Joker (2019)                       1479
Interstellar (2014)                1386
1917 (2019)                         820
Avengers: Endgame (2019)            812
The Shawshank Redemption (1994)     707
Gravity (2013)                      653
The Wolf of Wall Street (2013)      581
Hacksaw Ridge (2016)                570
Avengers: Infinity War (2018)       535
La La Land (2016)                   510
Name: Título, dtype: int64


Podemos observar que o Coringa tem sido o queridinho do público. São 1479 notas 10.